File: codes/javascript/src/004_loops.js
const generateNumberSequence = (startNumber, stopNumber) => {
    const numberSequenceArray = [];
    if (stopNumber > startNumber) {
        for (let anyNumber = startNumber; (anyNumber <= stopNumber); anyNumber += 1) {
            numberSequenceArray.push(anyNumber);
        }
        return numberSequenceArray;
    }
    if (startNumber > stopNumber) {
        for (let anyNumber = startNumber; (anyNumber >= stopNumber); anyNumber -= 1) {
            numberSequenceArray.push(anyNumber);
        }
        return numberSequenceArray;
    }
    return numberSequenceArray;
};

// while loop

{
    let i = 1;
    while (true) {
        if (i === 2) {
            if (i >= 5) break;
            i += 1;
            continue;
        }
        console.log(`while loop v1, i is ${i}`);
        if (i >= 5) break;
        i += 1;
    }
}

{
    let i = 1;
    while (i <= 5) {
        if (i === 2) {
            i += 1;
            continue;
        }
        console.log(`while loop v2, i is ${i}`);
        i += 1;
    }
}

// for loop

for (let i = 1; i <= 10; i += 1) {
    if (i > 5) break;
    if (i === 2) continue;
    console.log(`for loop, i is ${i}`);
}

// forEach loop

generateNumberSequence(1, 10).forEach((i) => {
    if (i > 5) return;
    if (i === 2) return;
    console.log(`forEach loop v1 ascending, i is ${i}`);
});

generateNumberSequence(10, 1).forEach((i) => {
    if (i <= 5) return;
    if (i === 9) return;
    console.log(`forEach loop v1 descending, i is ${i}`);
});

File: codes/csharp/004_loops.cs
using System; // Console, Func<>, Action<>, Action
using System.Linq; // Enumerable

class Program {
    static void Main(string[] Args) {
        // while loop

        int i1 = 1;
        while (true) {
            if (i1 == 2) {
                if (i1 >= 5) break;
                i1 += 1;
                continue;
            }
            Console.WriteLine($"while loop v1, i1 is {i1}");
            if (i1 >= 5) break;
            i1 += 1;
        }

        int i2 = 1;
        while (i2 <= 5) {
            if (i2 == 2) {
                i2 += 1;
                continue;
            }
            Console.WriteLine($"while loop v2, i2 is {i2}");
            i2 += 1;
        }

        // for loop

        for (int i3 = 1; i3 <= 10; i3 += 1) {
            if (i3 > 5) break;
            if (i3 == 2) continue;
            Console.WriteLine($"for loop, i3 is {i3}");
        }

        // forEach loop

        foreach (int i4 in Enumerable.Range(1, 10)) {
            if (i4 > 5) break;
            if (i4 == 2) continue;
            Console.WriteLine($"forEach loop ascending, i4 is {i4}");
        }

        foreach (int i5 in Enumerable.Range(1, 10).Reverse().ToList()) {
            if (i5 <= 5) break;
            if (i5 == 9) continue;
            Console.WriteLine($"forEach loop descending, i5 is {i5}");
        }
    }
}
